### PR TITLE
Add stake calculation tests with inflation/slashing

### PR DESCRIPTION
**Pull request by ryoqun**
#### Problem

There is no test for expected change of stakes after inflation and slashing with stable rate, which is fixed at the original activation/deactivation with stake_history at the time.

In other words, there is no test to change `self.stake` as far as I know... Inflation and slashing directly mutates this. So, this isn't static state for the entire lifecycle of stake account.. Current set of tests implicitly tells otherwise. There should be at least one such a test.

#### Summary of Changes

~~Add it before we break it. Still a bit dirty code I'm occupied with other fixes.~~ I mostly cleaned this up.

Fixes #

### CODE REVIEW

File: programs/stake/src/stake_state.rs [hunk 1]
```diff
@@ -1803,6 +1803,164 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_inflation_and_slashing_with_activating_stake() {
```
Comment (ryoqun) [inline programs/stake/src/stake_state.rs:1807]: todo: add an accompanying test for the case of deactivating accounts.

Comment (ryoqun) [reply programs/stake/src/stake_state.rs]: well, this test already contains deactivation!

**Comment by ryoqun** [comment]
@CriesofCarrots Could you glance at this at your convenient time?

**Review by CriesofCarrots — APPROVED** [review]
lgtm

**Pull request merged**